src/test/java/htsjdk/samtools/util/CigarUtilTest.java

```diff
@@ -67,46 +69,81 @@ public void basicTest(final String testName, final int start, final String input
            Assert.assertTrue(sizeChange >= 0, "sizeChange >= 0. " + sizeChange);
       }
        Assert.assertEquals (TextCigarCodec.encode(newCigar), expectedCigar, testName);
-       Assert.assertEquals(newCigar.getReadLength(), oldCigar.getReadLength());
+       if(clippingOperator == CigarOperator.SOFT_CLIP) {
+           Assert.assertEquals(newCigar.getReadLength(), oldCigar.getReadLength());
+       }
        Assert.assertNull(newCigar.isValid(testName, -1));
     }
 
     @DataProvider(name = "clipData")
     private Object[][] getCigarClippingTestData() {
```

Comment:
You need tests to check the bases clipped to make sure that we hardclipped the right bases.